File: March 26, 2026.py
def get_movie_night_cost(day, showtime, number_of_tickets):

    weekend = ["Friday", "Saturday", "Sunday"]
    weekday = ["Monday", "Wednesday", "Thursday"]
    [h, m] = showtime.split(":")
    hour = int(h)
    minute = int(m[:2])
    period = m[2:]


    if period == "pm" and hour != 12:
        hour += 12

    total_minutes = (hour * 60) + minute

    if day == "Tuesday":
        total_cost = number_of_tickets * 5
        return f"${total_cost}.00"

    if day in weekend:
        total_cost = number_of_tickets * 12

    if day in weekday:
        total_cost = number_of_tickets * 10

    if total_minutes <= (17 * 60):
            total_cost -= number_of_tickets * 2

    return f"${total_cost}.00"

File: March 26, 2026.js
function getMovieNightCost(day, showtime, numberOfTickets) {

  const weekend = ["Friday", "Saturday", "Sunday"];
  const weekday = ["Monday", "Wednesday", "Thursday"];
  const [hourPart, minutePart] = showtime.split(":");
  let hour = parseInt(hourPart);
  const minute = parseInt(minutePart.slice(0, 2));
  const period = minutePart.slice(2);

  if (period === "pm" && hour !== 12) {
    hour += 12;
  }

  const totalMinutes = (hour * 60) + minute;
  let totalCost = 0;

  if (day == "Tuesday") {
    totalCost = numberOfTickets * 5;
      return `$${totalCost.toFixed(2)}`
  }
        

  if (weekend.includes(day)) {
    totalCost = numberOfTickets * 12;
  }
        

  if (weekday.includes(day)) {
    totalCost = numberOfTickets * 10;
  }
        

  if (totalMinutes <= (17 * 60)) {
    totalCost -= numberOfTickets * 2;
  }

  return `$${totalCost.toFixed(2)}`;
}
